### PR TITLE
Feature/robot  asc pom

### PR DESCRIPTION
**Pull request by skristem**
Convert Automated Soft Credits tests to POM and fix unstable tests
# Critical Changes

# Changes
# Issues Closed

# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: robot/Cumulus/resources/OpportunityContactRolePageObject.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+#         Waits for the current page to be a Data Import list view
+#         """
+#         self.selenium.wait_until_location_contains("/list",timeout=60, message="Records list view did not load in 1 min")
+#         self.selenium.location_should_contain("DataImport__c",message="Current page is not a DataImport List view")
+            
+    def _wait_to_appear(self, timeout=None):
+        """This function is called by the keyword 'wait for page object to appear'.
+        Custom page objects can override this to verify that the object is visible.
+        """
+        self.selenium.wait_until_location_contains("OpportunityContactRoles/view",timeout=60, message="Related list view did not load in 1 min")
```
Comment (sgovindankutty) [inline robot/Cumulus/resources/OpportunityContactRolePageObject.py]: Any reason to hardcode timeout here?

File: robot/Cumulus/resources/NPSP.py [hunk 2]
```diff
@@ -1397,6 +1398,21 @@ def edit_record_field_value(self,field,value):
...
+        scroll_loc=npsp_lex_locators["span_button"].format(field)
+        # To make sure the field we want to edit has rendered
+        # and is not obscured by the footer, scroll down a little below the element
+        self.selenium.scroll_element_into_view(scroll_loc)
+        self.selenium.execute_javascript("window.scrollBy(0,50)")
+        btn="Edit "+field
+        self.selenium.click_button(btn)
+        footer=npsp_lex_locators["record"]["footer"]
+        self.selenium.wait_until_page_contains_element(footer)
+        time.sleep(2)
```
Comment (rjanjanam) [inline robot/Cumulus/resources/NPSP.py]: Is this sleep needed here or can be replaced ?

File: robot/Cumulus/resources/OpportunityContactRolePageObject.py (hunk 1 above)
Comment (skristem) [reply robot/Cumulus/resources/OpportunityContactRolePageObject.py]: on metaci it sometimes takes upto a minute depending on slowness. This timeout does not impact if page loads quickly as far as I noticed

File: robot/Cumulus/resources/NPSP.py (hunk 2 above)
Comment (skristem) [reply robot/Cumulus/resources/NPSP.py]: yes I tried many options but otherthan hard coded sleep, I couldn't get this working most of the time.

File: robot/Cumulus/tests/browser/automated_soft_credits/asc_primary_and_affiliation.robot [hunk 3]
```diff
@@ -2,43 +2,70 @@
...
 ...             robot/Cumulus/resources/NPSPSettingsPageObject.py
-Suite Setup     Open Test Browser
+Suite Setup     Run keywords
+...             Open Test Browser
+...             Setup Test Data
 Suite Teardown  Delete Records and Close Browser
 
+***Keywords***
+Setup Test Data
+    Setupdata           account      None    None    ${account_fields}
```
Comment (rjanjanam) [inline robot/Cumulus/tests/browser/automated_soft_credits/asc_primary_and_affiliation.robot]: We can replace Nones, by specifying something like
Setupdata   account   account_data=${account_fields}     
and say, if you want to create an opportunity associated with that account within the same step
Setupdata   account   account_data=${account_fields}    opportunity_data=${opportunity_fields}

File: robot/Cumulus/tests/browser/automated_soft_credits/asc_reciprocal_relation.robot [hunk 4]
```diff
@@ -2,69 +2,93 @@
...
+    
+    Go To Page                              Details                              Contact                                
+    ...                                     object_id=${data}[contact4][Id]
     Select Tab  Related
-    Select Relatedlist    Contact Roles
+    Click Related Table Item Link           Relationships                        ${data}[contact3][FirstName] ${data}[contact3][LastName]
+    Edit Record Dropdown Value             	Related Opportunity Contact Role     Soft Credit
+    Save Record
+    &{opportunity2} =                       API Create Opportunity               ${data}[contact3][AccountId]    Donation    
+    ...                                     Name=Reciprocal test $500 donation
```
Comment (rjanjanam) [inline robot/Cumulus/tests/browser/automated_soft_credits/asc_reciprocal_relation.robot]: this can be in setup ?

Comment (skristem) [reply robot/Cumulus/tests/browser/automated_soft_credits/asc_reciprocal_relation.robot]: no it cant be in setup. Other stuff has to happen before this opp can be created

File: robot/Cumulus/tests/browser/automated_soft_credits/asc_primary_and_affiliation.robot (hunk 3 above)
Comment (skristem) [reply robot/Cumulus/tests/browser/automated_soft_credits/asc_primary_and_affiliation.robot]: I tried Setupdata account account_data=${account_fields} and it didn't work, was giving me errors so had to go with this as last option.

Comment (skristem) [reply robot/Cumulus/tests/browser/automated_soft_credits/asc_primary_and_affiliation.robot]: And opportunity cannot be created together until the stuff is created so cant do it together. Also was having issue with opp created for acct but noticed u added code in ur PR for that so didn't add in mine.

File: robot/Cumulus/resources/NPSP.py [hunk 5]
```diff
@@ -511,6 +511,7 @@ def verify_occurence(self,title,value):
            
         
     def check_record_related_item(self,title,value):
+        self.salesforce.load_related_list(title)
```
Comment (sgovindankutty) [inline robot/Cumulus/resources/NPSP.py]: can you add documentation to this keyword?

Comment (skristem) [reply robot/Cumulus/resources/NPSP.py]: ahh missed it, will add

**Pull request merged**